### PR TITLE
fix: remove "managed-by" fixing helm conflict (fixes #6421)

### LINKED ISSUE #6421: [Helm] "app.kubernetes.io/managed-by" must equal "Helm": current value is "skaffold"
### Expected behavior

`skaffold run` succeeds multiple times.

### Actual behavior

```
> skaffold run
...
Error: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "foobar" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: key "app.kubernetes.io/managed-by" must equal "Helm": current value is "skaffold"
deploying "fn-drupal-operator": install: exit status 1

```

It appears the underlying helm execution cannot handle the modified "app.kubernetes.io/managed-by" value for some resources. This causes multiple executions of `skaffold run` to fail in some cases.

It seems the resources it struggles with are core k8s resources in the default namespace. This is just a guess, but we have many resources that it doesn't have any issue with.

##### Workaround

```bash
kubectl api-resources --verbs=list \
  | grep 'k8s\.io' | cut -d' ' -f1 \
  | xargs -P8 -n 1 kubectl get --show-kind --ignore-not-found -l app.kubernetes.io/managed-by=skaffold -o name \
  | xargs kubectl delete
skaffold run
```

### Information
```
helm version
version.BuildInfo{Version:"v3.6.3", GitCommit:"d506314abfb5d21419df8c7e7e68012379db2354", GitTreeState:"dirty", GoVersion:"go1.16.5"}
```

- Skaffold version: v1.29.0
- Operating system: macOS 11.5
- Installed via: Homebrew
- Contents of skaffold.yaml:

TBD; too much to put here

### Steps to reproduce the behavior

1. a clonable repository with the sample skaffold project
2. `skaffold <command>`
3. ...

I will try to reproduce this on an example I can share. So far I haven't been able to reproduce outside of an internal project.

### PR DESCRIPTION
**Pull request by glennpratt**
Fixes: #6421

**Description**
remove "managed-by" fixing helm conflict

**User facing changes (remove if N/A)**
remove "managed-by"

### CODE REVIEW

**Comment by MarlonGamez** [comment]
hey @glennpratt , thanks for opening this. It looks good! I think there is just a gofmt error that needs to be resolved 👍🏼 I'll approve and merge once that's all good

File: integration/render_test.go [hunk 1]
```diff
@@ -688,7 +687,6 @@ kind: Pod
 metadata:
   labels:
     app.kubernetes.io/managed-by: SOMEDYNAMICVALUE
-    skaffold.dev/run-id: SOMEDYNAMICVALUE
```
Comment (MarlonGamez) [inline integration/render_test.go:691]: didn't catch this at first but looks like this is the wrong line being removed here right?

**Review by MarlonGamez — COMMENTED** [review]
caught a small error with the updated integration test

File: integration/render_test.go (hunk 1 above)
Comment (glennpratt) [reply integration/render_test.go]: Oof, thanks for catching that. Fixed.

**Review by MarlonGamez — APPROVED** [review]
looks good to me 👍🏼

**Pull request merged**